torch/_refs/__init__.py

```diff
@@ -6332,21 +6332,15 @@ def _infer_scalar_type(obj):
 
 # Analogous to recursive_store
 # xref: recursive_store in torch/csrc/utils/tensor_new.cpp
-def _recursive_build(sizes, dim, scalarType, obj):
-    ndim = len(sizes)
-    assert dim <= ndim
-    if dim == ndim:
+def _recursive_build(scalarType, obj):
+    if isinstance(obj, Tensor) and obj.ndim <= 1:
+        obj = obj.item()
```

Comment:
No need to go through item(), just .to(scalarType, copy=True)

Reply:
seems like `torch.stack` expects its entries to have the same size.

```
# tried this
-        obj = obj.item()
+        return obj.to(scalarType, copy=True)

# saw this
stack expects each tensor to be equal size, but got torch.Size([1]) at entry and torch.Size([]) at entry 1
```